src/container/container_iv.c

```diff
@@ -592,21 +643,268 @@ cont_iv_capability_update(void *ns, uuid_t cont_hdl_uuid, uuid_t cont_uuid,
+			break;
+		case DAOS_PROP_CO_LAYOUT_TYPE:
+			iv_prop->cip_layout_type = prop_entry->dpe_val;
+			break;
+		case DAOS_PROP_CO_LAYOUT_VER:
+			iv_prop->cip_layout_ver = prop_entry->dpe_val;
+			break;
+		case DAOS_PROP_CO_CSUM:
+			iv_prop->cip_csum = prop_entry->dpe_val;
+			break;
+		case DAOS_PROP_CO_REDUN_FAC:
+			iv_prop->cip_redun_fac = prop_entry->dpe_val;
+			break;
+		case DAOS_PROP_CO_REDUN_LVL:
+			iv_prop->cip_redun_lvl = prop_entry->dpe_val;
+			break;
+		case DAOS_PROP_CO_SNAPSHOT_MAX:
+			iv_prop->cip_snap_max = prop_entry->dpe_val;
+			break;
+		case DAOS_PROP_CO_COMPRESS:
+			iv_prop->cip_compress = prop_entry->dpe_val;
+			break;
+		case DAOS_PROP_CO_ENCRYPT:
+			iv_prop->cip_encrypt = prop_entry->dpe_val;
+			break;
+		case DAOS_PROP_CO_ACL:
+			acl = prop_entry->dpe_val_ptr;
+			if (acl != NULL)
+				memcpy(&iv_prop->cip_acl, acl,
+				       daos_acl_get_size(acl));
+			break;
+		default:
+			D_ASSERTF(0, "bad dpe_type %d\n", prop_entry->dpe_type);
+			break;
+		}
+	}
+}
+
+static int
+cont_iv_prop_g2l(struct cont_iv_prop *iv_prop, daos_prop_t *prop)
+{
+	struct daos_prop_entry	*prop_entry;
+	struct daos_acl		*acl;
+	void			*label_alloc = NULL;
+	void			*acl_alloc = NULL;
+	int			 i;
+	int			 rc = 0;
+
+	D_ASSERT(prop->dpp_nr == CONT_PROP_NUM);
+	for (i = 0; i < CONT_PROP_NUM; i++) {
+		prop_entry = &prop->dpp_entries[i];
+		prop_entry->dpe_type = DAOS_PROP_CO_MIN + i + 1;
+		switch (prop_entry->dpe_type) {
+		case DAOS_PROP_CO_LABEL:
+			D_ASSERT(strlen(iv_prop->cip_label) <=
+				 DAOS_PROP_LABEL_MAX_LEN);
+			prop_entry->dpe_str = strdup(iv_prop->cip_label);
+			if (prop_entry->dpe_str)
+				label_alloc = prop_entry->dpe_str;
+			else
+				D_GOTO(out, rc = -DER_NOMEM);
+			break;
+		case DAOS_PROP_CO_LAYOUT_TYPE:
+			prop_entry->dpe_val = iv_prop->cip_layout_type;
+			break;
+		case DAOS_PROP_CO_LAYOUT_VER:
+			prop_entry->dpe_val = iv_prop->cip_layout_ver;
+			break;
+		case DAOS_PROP_CO_CSUM:
+			prop_entry->dpe_val = iv_prop->cip_csum;
+			break;
+		case DAOS_PROP_CO_REDUN_FAC:
+			prop_entry->dpe_val = iv_prop->cip_redun_fac;
+			break;
+		case DAOS_PROP_CO_REDUN_LVL:
+			prop_entry->dpe_val = iv_prop->cip_redun_lvl;
+			break;
+		case DAOS_PROP_CO_SNAPSHOT_MAX:
+			prop_entry->dpe_val = iv_prop->cip_snap_max;
+			break;
+		case DAOS_PROP_CO_COMPRESS:
+			prop_entry->dpe_val = iv_prop->cip_compress;
+			break;
+		case DAOS_PROP_CO_ENCRYPT:
+			prop_entry->dpe_val = iv_prop->cip_encrypt;
+			break;
+		case DAOS_PROP_CO_ACL:
+			acl = &iv_prop->cip_acl;
+			if (acl->dal_len > 0) {
+				D_ASSERT(daos_acl_validate(acl) == 0);
+				prop_entry->dpe_val_ptr =
+					daos_acl_dup(acl);
+				if (prop_entry->dpe_val_ptr)
+					acl_alloc = prop_entry->dpe_val_ptr;
+				else
+					D_GOTO(out, rc = -DER_NOMEM);
+			} else {
+				prop_entry->dpe_val_ptr = NULL;
+			}
+			break;
+		default:
+			D_ASSERTF(0, "bad dpe_type %d\n", prop_entry->dpe_type);
+			break;
+		}
+	}
+
+out:
+	if (rc) {
+		if (acl_alloc)
+			D_FREE(acl_alloc);
```

Comment:
I think this should be daos_acl_free not D_FREE Kris will know better but she's out on sabbatical till the end of the month.

Reply:
agreed, although it is same now, I'll refresh it.